components/browser/session/src/main/java/mozilla/components/browser/session/SessionManager.kt

```diff
@@ -173,18 +181,31 @@ class SessionManager(
         }
 
         selectedIndex = NO_SELECTION
-        notifyObservers { onAllSessionsRemoved() }
+
+        if (defaultSession != null) {
+            add(defaultSession.invoke())
+        } else {
+            notifyObservers { onAllSessionsRemoved() }
+        }
```

Comment:
That's the part I'm not sure about: An observer would never see the removal, just an add. Why wouldn't we first notify about removal and then add the new session?

Reply:
We could! With a default session present, we're never really in an "empty" state that's observable to the outside world - that's why I didn't notify observers. Reflecting on the docstring of "all sessions removed" callback, it seems that the intent there isn't to indicate an empty manager, but to indicate that one of the blanket remove* functions were invoked. That interpretation would dictate that yeah, they should be notified here.

Reply:
ISTM that the observer "onAllSessionsRemoved" method should be named differently, since it doesn't actually fire when "all sessions are removed". Filed a follow up #948.